library/ssl_srv.c

```diff
@@ -3048,8 +3049,8 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
 
         MBEDTLS_SSL_DEBUG_MSG( 2, ( "ECDHE curve: %s", (*curve)->name ) );
 
-        if( ( ret = mbedtls_ecdh_setup( &ssl->handshake->ecdh_ctx,
-                                        (*curve)->grp_id ) ) != 0 )
+        if( ( ret = mbedtls_ecp_group_load( &ssl->handshake->ecdh_ctx.grp,
```

Comment:
@sbutcher-arm On first sight, this change seems unrelated to the problem this PR sets out to fix. Is it intentional?

Reply:
That looks like a rather curious mistake in the rebase.